src/test/java/lambda/part2/exercise/Exercise1.java

```diff
@@ -30,35 +28,37 @@ void sameAgesCheckerUsingBiPredicate() {
         Person person2 = new Person("Дмитрий", "Гущин", 33);
         Person person3 = new Person("Илья", "Жирков", 22);
 
-        // TODO create variable sameAgesChecker: (Person, Person) -> boolean, using BiPredicate
-
-//        assertThat(sameAgesChecker.test(person1, person2), is(true));
-//        assertThat(sameAgesChecker.test(person1, person3), is(false));
-//        assertThat(sameAgesChecker.test(person2, person3), is(false));
+        //create variable sameAgesChecker: (Person, Person) -> boolean, using BiPredicate
+        BiPredicate<Person, Person> sameAgesChecker = (person1_1, person2_2) -> person1.getAge() == person2_2.getAge();
 
-        // FIXME remove after implementation
-        throw new UnsupportedOperationException("Not implemented");
+        assertThat(sameAgesChecker.test(person1, person2), is(true));
+        assertThat(sameAgesChecker.test(person1, person3), is(false));
+        assertThat(sameAgesChecker.test(person2, person3), is(false));
     }
 
     // TODO создать метод getFullName: Person -> String, извлекающий из объекта Person строку в формате "имя фамилия".
-    // private static ... getFullName(...) {
+    private static String getFullName(Person person) {
+        return person.getFirstName() + " " + person.getLastName();
+    }
 
-    // TODO создать метод createExtractorAgeOfPersonWithTheLongestFullName: (Person -> String) -> ((Person, Person) -> int),
-    // TODO - принимающий способ извлечения полного имени из объекта Person
-    // TODO - возвращающий BiFunction, сравнивающий два объекта Person и возвращающий возраст того, чье полное имя длиннее.
-    // private static ... createExtractorAgeOfPersonWithTheLongestFullName(...) {
+    // создать метод createExtractorAgeOfPersonWithTheLongestFullName: (Person -> String) -> ((Person, Person) -> int),
+    // - принимающий способ извлечения полного имени из объекта Person
+    // - возвращающий BiFunction, сравнивающий два объекта Person и возвращающий возраст того, чье полное имя длиннее.
+    private static BiFunction<Person, Person, Integer> createExtractorAgeOfPersonWithTheLongestFullName(Function<Person, String> getFullNameFunction) {
+        return (person1, person2) -> getFullName(person1).length() >= getFullName(person2).length() ? person1.getAge() : person2.getAge();
+    }
```

Comment:
You didn't use in the method's body getFullNameFunction

Reply:
Thank you for recommendation. I have done it.